src/finch/tensor.py

```diff
@@ -86,7 +87,7 @@ def __init__(
         *,
         fill_value: np.number = 0.0,
     ):
-        if isinstance(obj, (int, float, complex, bool)):
+        if isinstance(obj, (int, float, complex, bool, list)):
```

Comment:
I'd avoid hard-coding list and check for `Iterable`.

Reply:
But that would be incorrect. We only support iterables that can be passed to `np.array(...)` to create a valid array.

`dict` and `set` are out of support (only `list` and `tuple`):
```python
In [9]: isinstance(set((1,2,3)), Iterable)
Out[9]: True

In [10]: np.array(set((1,2,3)))
Out[10]: array({1, 2, 3}, dtype=object)

In [12]: isinstance({1:2}, Iterable)
Out[12]: True

In [13]: np.array({1:2})
Out[13]: array({1: 2}, dtype=object)
```